File: C++/1700.cpp
class Solution {
public:
    int countStudents(vector<int>& students, vector<int>& sandwiches) {
        int ans = students.size();
        vector<int> count = {0, 0};
        for(int student : students) {
            count[student]++;
        }
        for(int sandwich : sandwiches) {
            if (count[sandwich] == 0) break;
            count[sandwich]--;
            ans--;
        }
        return ans;
    }
};

// Topic: Array
// Time Complexity: O(n)
// Space Complexity: O(1)

File: Python/1700.py
from typing import List
from collections import Counter

class Solution:
    def countStudents(self, students: List[int], sandwiches: List[int]) -> int:
        ans = len(students)
        count = Counter(students)

        for s in sandwiches:
            if count[s] == 0:
                break
            count[sandwiches[0]] -= 1
            ans -= 1

        return ans

# Topic: Array
# Time Complexity: O(n)
# Space Complexity: O(1)